app/models/spree/payment_method/stripe_credit_card.rb

```diff
@@ -81,6 +81,21 @@ def void(response_code, _creditcard, _transaction_options)
         gateway.void(response_code, {})
       end
 
+      def payment_intents_refund_reason
+        Spree::RefundReason.where(name: Spree::Payment::Cancellation::DEFAULT_REASON).first_or_create
+      end
+
+      def try_void(payment)
+        if v3_intents? && payment.completed?
```

Comment:
Why don't you just return `false` here? It seems like solidus already handles the refund [here](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment/cancellation.rb#L30-L36) 🤔

Reply:
@ChristianRimondi that way the payment would remain to its original state, instead of becoming`void`, as this [line](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment/cancellation.rb#L32) would not run. That line executes [this method](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment/processing.rb#L190).